src/templates/static.tsx

```diff
@@ -143,16 +147,28 @@ const components = {
   YouTube,
 }
 
-const StaticPage = ({ data: { siteData, pageData: mdx }, pageContext }) => {
+const StaticPage = ({
+  data: { siteData, pageData: mdx },
+  pageContext,
+}: PageProps<Queries.StaticPageQuery, Context>) => {
   const intl = useIntl()
-  const isRightToLeft = isLangRightToLeft(mdx.frontmatter.lang)
 
-  const lastUpdatedDate = mdx.parent.fields
-    ? mdx.parent.fields.gitLogLatestDate
-    : mdx.parent.mtime
+  if (!siteData || !mdx?.frontmatter || !mdx.parent) {
+    throw new Error(
+      "Static page template query does not return expected values"
+    )
+  }
+
+  const isRightToLeft = isLangRightToLeft(mdx.frontmatter.lang as Lang)
+
+  // FIXME: remove this any, currently not sure how to fix the ts error
```

Comment:
Hmmm what was the TS error? Is the type inconsistent?

Not sure if the comment below this is related:
```
// TODO some `gitLogLatestDate` are `null` - why?
```

Reply:
So, the auto-generated types for `mdx` are kind of weird. I have an error with in the next line where we call `parent.fields`.

The generated type for `mdx` is saying that it could possibly be `{}`, not sure if that is correct.

The error:
```
Property 'fields' does not exist on type '{} | { readonly mtime: string; readonly fields: { readonly gitLogLatestDate: string | null; } | null; }'.
  Property 'fields' does not exist on type '{}'.
```